charts/istio/istio-istiod/templates/autoscale.yaml

```
@@ -1,19 +1,20 @@
-apiVersion: {{ include "hpaversion" . }}
-kind: HorizontalPodAutoscaler
+apiVersion: autoscaling.k8s.io/v1beta2
+kind: VerticalPodAutoscaler
 metadata:
   name: istiod
   namespace: {{ .Release.Namespace }}
   labels:
 {{ .Values.labels | toYaml | indent 4 }}
 spec:
-  maxReplicas: 5
-  minReplicas: 2
-  scaleTargetRef:
+  targetRef:
     apiVersion: {{ include "deploymentversion" . }}
     kind: Deployment
     name: istiod
-  metrics:
-  - type: Resource
-    resource:
-      name: cpu
-      targetAverageUtilization: 80
+  updatePolicy:
+    updateMode: Auto
+  resourcePolicy:
+    containerPolicies:
+      - containerName: '*'
```

I do not see any other container in the deployment, thus you can specify it explicitly.
```suggestion
      - containerName: 'discovery'
```